### PR TITLE
Rename theme.palette.type for mui4 to mui5

### PR DESCRIPTION
**Pull request by cipher-tech**
renamed theme-palette-type for mui4 to mui5.

### CODE REVIEW

File: .grit/patterns/mui4_to_mui5.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+This pattern Rename theme.palette.type in MUI v4 to MUI v5
+
+tags: #react, #migration, #complex
+
+```grit
+engine marzano(0.1)
+language js
+
+// Most of the logic for this pattern is in mui4_to_mui5.grit
+// https://github.com/getgrit/js/blob/main/.grit/patterns/mui4_to_mui5.grit
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:12]: This file does not exist. Don't link to files that don't exist.

File: .grit/patterns/mui4_to_mui5.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+`createTheme({ palette: { type: $arg } })` => `createTheme({ palette: { mode: $arg } })` where {
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:14]: This pattern is not sufficiently general. Ex. this won't match a theme that has anything except a palette.

Here's how you would generalize it more.
```
`createTheme($theme)` where {
  $theme <: contains `palette: { $palette }`,
  $palette <: contains `type: $arg` => `type: $arg`
}
```

File: .grit/patterns/mui4_to_mui5.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  $arg <: within `palette: { $_ }`
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:15]: You don't need to check this - it's already inferred from the context.

File: .grit/patterns/mui4_to_mui5.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+}
+```
+
+## Replace Rename theme.palette.type in MUI v4 to MUI v5
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:19]: Please include at least 3 test cases.

**Review by morgante — CHANGES_REQUESTED** [review]

File: .grit/patterns/mui4_to_mui5.md (hunk 4 above)
Comment (morgante) [reply .grit/patterns/mui4_to_mui5.md:17]: @cipher-tech Put each test case under its own heading, with a title of what it tests.

File: .grit/patterns/mui4_to_mui5.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+engine marzano(0.1)
+language js
+
+`createTheme($theme)` where {
+  $theme <: contains `palette: { $palette }`,
+  $palette <: contains `type: $arg` => `mode: $arg`
+}
+```
+
+## Rename type to mode for palette > MUI v4 to MUI v5
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:17]: You don't need to include the context on every test name. The document structure already captures that.

```suggestion
## Rename type to mode for palette
```

File: .grit/patterns/mui4_to_mui5.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
+## title: Convert Rename theme.palette.type In MUI v4 To MUI v5
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:1]:
```suggestion
## title: Upgrade MUI v4 to MUI v5
```

File: .grit/patterns/mui4_to_mui5.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
+## title: Upgrade MUI v4 to MUI v5
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:1]: This title prefix is incorrect.

File: .grit/patterns/mui4_to_mui5.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+This pattern Rename theme.palette.type in MUI v4 to MUI v5
```
Comment (morgante) [inline .grit/patterns/mui4_to_mui5.md:3]: Please pay attention to grammar.

File: .grit/patterns/mui4_to_mui5.md (hunk 1 above)
Comment (cipher-tech) [reply .grit/patterns/mui4_to_mui5.md]: This has resolved @morgante

File: .grit/patterns/mui4_to_mui5.md (hunk 2 above)
Comment (cipher-tech) [reply .grit/patterns/mui4_to_mui5.md]: This has been updated @morgante

File: .grit/patterns/mui4_to_mui5.md (hunk 3 above)
Comment (cipher-tech) [reply .grit/patterns/mui4_to_mui5.md]: This has been updated

File: .grit/patterns/mui4_to_mui5.md (hunk 4 above)
Comment (cipher-tech) [reply .grit/patterns/mui4_to_mui5.md]: I have included test headers.

File: .grit/patterns/mui4_to_mui5.md (hunk 7 above)
Comment (cipher-tech) [reply .grit/patterns/mui4_to_mui5.md]: I have updated the title.

**Review by morgante — APPROVED** [review]

**Pull request merged**